enterprise/api_client/discovery.py

```diff
@@ -411,3 +420,13 @@ def __init__(self):
                 raise ImproperlyConfigured(_("The configured CatalogIntegration service user does not exist."))
         else:
             raise ImproperlyConfigured(_("There is no active CatalogIntegration."))
+
+    @classmethod
```

Comment:
I don't think you want the `@classmethod` decorator here.

Reply:
I wanted to encapsulate handling the `ImproperlyConfigured` error into this function if possible to make the code in our views easy to read. For that, I needed to instantiate it here. Or do you mean make it a static method?

Reply:
Ah, I see now. Makes sense. Thanks for the clarification.